thoth-api/migrations/0.9.13/up.sql

```diff
@@ -0,0 +1,78 @@
+-- Add work table field to track when any of the work's relations were last updated.
+
+ALTER TABLE work
+    ADD COLUMN relation_updated_at TIMESTAMP NOT NULL DEFAULT '1970-01-01 00:00:00';
```

Comment:
I'm not convinced by the default value. It shouldn't be too tricky to get the current last updated timestamp as the original value. You can first add the column as `NULL`, find the right timestamp, and then convert the column to `NOT NULL` with `CURRENT_TIMESTAMP` as the `DEFAULT`.

To find the current value, you can use `GREATEST()`, as in this **not complete** example:

```
SELECT w.work_id, GREATEST(w.updated_at, c.updated_at, f.updated_at, i.updated_at, l.updated_at, p.updated_at, r.updated_at, s.updated_at) updated
FROM work w
  LEFT JOIN contribution c USING (work_id)
  LEFT JOIN funding f USING (work_id)
  LEFT JOIN issue i USING (work_id)
  LEFT JOIN language l USING (work_id)
  LEFT JOIN publication p USING (work_id)
  LEFT JOIN reference r USING (work_id)
  LEFT JOIN subject s USING (work_id)
GROUP BY w.work_id, updated
;
```

Reply:
The inclusion of `w.updated_at` in the example suggests that a work's `relation_updated_at` field should also be updated whenever the base record itself is updated. The PR as currently written doesn't do this. Do you want me to make that change?

Reply:
Yes, I think it'd be useful, to have a field that reflects if the work record (including all its relationships) has changed. Perhaps the name `relation_updated_at` doesn't fully reflect that (but I can't think of a different one)

Reply:
OK, I can incorporate that into the main logic. But as for the default value, I've just realised that all works' `updated_at` values seem to get updated when we run the migration (since this is changing every work record by adding a new column to it). So the output of the very intricate `GREATEST()` calculation will always just come out as the timestamp of the migration.

Not sure what to do about that...